### ISSUE TITLE
Statistics interpretation and values

### ISSUE THREAD

**Issue by mebbert** [ksahlin/BESST#40]
Krisoffer,

I appreciate your excellent support for BESST. BESST has been very easy to use. Going through the statistics file, it's difficult to know what some of the statistics are measuring and what is a normal range. Can you provide some insight? It would also be helpful to clarify units in the stats file.

Can you tell what unit these statistics are in (i.e., the mean what?):
> PASS 1
Mean before filtering : 2815.5728
Std_est  before filtering:  730.766563077
Mean converged: 2805.22808404
Std_est converged:  479.316095255
Skewness of distribution:  0.392962285269

What is a reasonable range for each of these statistics?:
> Contamine mean before filtering : 471.451245949
Contamine stddev before filtering:  467.073849795
Contamine mean converged: 413.536238767
Contamine std_est converged:  156.57484321

What is a resonable range for these statistics (I removed the ones I understand)? This is for a 3kb MP library. Why is number of required links `None`? It's set to `5` in your example output.
> LIBRARY STATISTICS
MP library PE contamination:
Contamine rate (rev comp oriented) estimated to:  0.434451909632
lib contamine mean (avg fragmentation size):  413.536238767
lib contamine stddev:  156.57484321
Number of contamined reads used for this calculation:  216881.0
-T (library insert size threshold) set to:  5683.87475718
-k set to (Scaffolding with contigs larger than):  4724.65091171
Number of links required to create an edge:  None
Maximum identical contig-end overlap-length to merge of contigs that are adjacent in a scaffold:  200

What is a reasonable range for these? I made comments on some of these:
> Parsing BAM file...
Number of non unique reads (at least one read non-unique in read pair) that maps to different contigs (filtered out from scaffolding):  5639881  # Seems really high
Reads with too large insert size from "USEFUL READS" (filtered out):  593154  # Seems high
Initial number of edges in G (the graph with large contigs):  128066
Initial number of edges in G_prime (the full graph of all contigs before removal of repats):  1900379
Number of duplicated reads indicated and removed:  114639  # Seems high
Mean coverage before filtering out extreme observations =  5.94613850826  # Seems very low, but I guess that's what we have. Do you think deeper coverage would make a big difference?
Std dev of coverage before filtering out extreme observations=  9.12324298951
Mean coverage after filtering =  5.90101135903
Std coverage after filtering =  0.711401219924
Length of longest contig in calc of coverage:  610609
Length of shortest contig in calc of coverage:  10575
Detecting repeats..
Removed a total of:  33821  repeats. With coverage larger than  9.66781905828
Number of edges in G (after repeat removal):  127699
Number of edges in G_prime (after repeat removal):  1637775
Number of BWA buggy edges removed:  0
Number of edges in G (after filtering for buggy flag stats reporting):  127699
Number of edges in G_prime  (after filtering for buggy flag stats reporting):  1637775

Finally, if I provided three MP libraries, is output from Pass3 the final output?

Thanks!

**Comment by ksahlin** [comment]
Thanks, I understand that it's a bit cryptic. I might make it more clear in later versions.
 
* Thats just the parameters to infer the shape of the insert size distribution. There is no "reasonable range" It depends on how the library is constructed.  [this section](https://github.com/ksahlin/BESST#my-mate-pair-libraries-contains-paired-end-contamination-what-should-i-do) shows two different libraries with very different ranges.
* None means that parameter -e is not set, so BESST infers in dynamically based on the graph complexity in the region.
* These numbers doesn't look stranger than what I've seen before. Yes, the non unique reads looks a little high, but from your N50 I assume that you are dealing with a pretty complex assembly. The rest of the numbers looks completely fine. Coverage of 6 is ok given that not too much is contributed from PE contamination, because span coverage will be high. What is the fraction of PE contamination (should be printed somewhere close to the lines you pasted)?
* The final output is in passX if there are X libraries. So yes, 3 in your case.

**Comment by mebbert** [comment]
Thanks Kristoffer,

The only output I see that mentions `PE` is the part I included previously. Can you tell me exactly what text I should look for?

**Comment by ksahlin** [comment]
this line
`Contamine rate (rev comp oriented) estimated to: `

**Comment by mebbert** [comment]
Ok, I included that in my original post:

> Contamine rate (rev comp oriented) estimated to: 0.434451909632

That's double what yours was, but I also used reads that were pre-trimmed. i.e., I didn't use the nxTrim to separate.

How would you proceed in my shoes?

Thanks!

**Comment by ksahlin** [comment]
ok, so you have about 3.3x coverage on average from mate pairs then ( (1.0-0.43)*5.9x ). Thats still decent. Based on your library stats I think you could scaffold the way the library is now (i.e. including all reads). However, It could be good for comparison to scaffold with the MP + Unknown from NxTrim. That is, filtering out the paired end contamination.

**Comment by mebbert** [comment]
Thanks, I'll give it a shot and report back.

I also should have included the final L50 and N50: 
>L50:  415 N50:  1510152 Initial contig assembly length:  2332561304

**Comment by mebbert** [comment]
Ok, after using NxTrim and scaffolding with only the MP + Unknown reads, the new stats are as follows:

>L50:  463 N50:  1365304 Initial contig assembly length:  2332561304

The question is whether the scaffold with the PE reads is really better (i.e., more accurate). I know we'd need to really dig into the genome and interrogate it, but I'm curious what you think from your experience.

Thanks for all of your help!

**Comment by ksahlin** [comment]
We found that PE contamination could help quite a lot for scaffolding fragmented assemblies in our [latest paper](http://bioinformatics.oxfordjournals.org/content/32/13/1925) of BESST, so it comes as no surprise. We saw an improvement especially in the arabidopsis data, where both the contigs and the MP library have similar characteristics as yours: fragmented assembly and fairly narrow MP isize distribution (results are in the supplementary data). However, there is always a trade off between how much is scaffolded and number of errors. Furthermore, what constitutes an error is also difficult to judge, because they will come in all sizes and forms, so it depends on how stringent one wants to be and what is the downstream purpose.

I think that the assembly with PE contamination will have shorter gaps in scaffolds at least, because the PE contamination helps especially with placing shorter sequences within gaps that the MP reads leaves in scaffolds.
